lib/viewmodels/map/map_pin_view_model.dart

```diff
@@ -34,6 +42,13 @@ class MapPinViewModel extends AutoDisposeAsyncNotifier<List<MapPinDetails>> {
     }
   }
 
+  BuildContext? context;
```

Comment:
I don't think this fits here. The `BuildContext` should not be passed in parameter to any function, even less to a view-model (a `BuildContext` is and should stay on the ui layer).

I think this file should be refactored by removing the idea of callback from the pins, but replacing it with parameters such as a `MapPopUpDetails`, which would contain a `title`, a nullable `content`, and a nullable function to call when the user taps on the button (if the function is null, the button is not displayed).

You should be able to reuse most of your code, so I'll still make comments on the code of this file (even if most of it will probably move).

Reply:
fixed with [994d03c](https://github.com/ProximaEPFL/proxima/pull/335/commits/994d03c1af074c06ac3ec0bd4f9b9c8c5ff4534d)